pandas/_libs/interval.pyx

```diff
@@ -82,12 +110,21 @@ cdef class Interval(IntervalMixin):
     Parameters
     ----------
     left : value
-        Left bound for the interval
+        Left bound for the interval.
     right : value
-        Right bound for the interval
-    closed : {'left', 'right', 'both', 'neither'}, default 'right'
+        Right bound for the interval.
+    closed : {'right', 'left', 'both', 'neither'}, default 'right'
         Whether the interval is closed on the left-side, right-side, both or
-        neither
+        neither.
+        A closed interval (in mathematics denoted by square brackets)
+        contains the edge points, i.e. the closed interval `[1, 5]` consists
+        of the points `1, 2, 3, 4, 5`.
+        An open interval (in mathematics denoted by parentheses) does not
+        contain the edge points, i.e. the open interval `(1, 5)` consists of
+        the points 2, 3, 4.
+        Intervals can also be half-open or half closed, i.e.
+        `[1,5) = 1, 2, 3, 4` and `(1, 5] = 2, 3, 4, 5`. See also the examples
+        section below.
```

Comment:
For each of the four examples above, it might be nice to explicitly say the corresponding pandas closed value, e.g `(1, 5]` corresponds to `'right'`.